lib/matplotlib/lines.py

```diff
@@ -1440,6 +1440,34 @@ def draw(self, renderer):
         super().draw(renderer)
 
 
+class _AxLineTransAxes(_AxLine):
+    """
+    A helper class that implements `~.Axes.axline_transaxes`, by recomputing
+    the artist transform at draw time.
+    """
+
+    def __init__(self, xy1, slope, **kwargs):
+        super().__init__([0, 1], [0, 1], **kwargs)
+        self._slope = slope
+        self._xy1 = xy1
+
+    def get_transform(self):
+        ax = self.axes
+        (vxlo, vylo), (vxhi, vyhi) = ax.transScale.transform(ax.viewLim)
+        x, y = (ax.transAxes + ax.transData.inverted()).transform(self._xy1)
```

Comment:
Wouldn't it be simpler to calculate the slope into axes coordinates and simply draw the line in axes co-ordinates?

Reply:
Good idea. I tried to stay close to the implementation of `axline` (as the original idea was to add this behavior into that function if a transform is passed, see #18625), but this may be an easier option.

Reply:
Hm, but it won't really get much simpler because we still need to somehow convert to data coordinates to calculate at which edges of the viewport (top/bottom or left/right) the ends of the line would be. That will of course depend on the current x/y limits.

Reply:
Wont the line clip if you just make it go from x_axes = -1 to 2?

Reply:
Ah, right, it does. I implemented a simpler solution in da040d0c68f5086ff0b939514a02d7d0dc0091e8, however it breaks the tests even though the plots visually seem to look the same. The difference seems to be very small, not sure why this is happening 😕

Reply:
You could probably safely up the tolerance if this is just roundoff from passing the positions through the transform stack...

```
@check_figures_equal(extensions=['png'], tol=0.19)
```